src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -684,6 +713,213 @@ IECore::DataPtr loadSpline(
+			}
+
+			InternedString adapterHandle;
+			int knotOffset;
+
+			// Insert a conversion shader to handle connection to component
+			auto inserted = currentSplineArrayAdapters.insert( { parameterName, std::make_pair( IECore::InternedString(), 0 ) } );
+			if( inserted.second )
+			{
+				IECore::InternedString splineValuesName = parameterName.string() + "Values";
+				auto findValues = newParameters.find( splineValuesName );
+				ShaderPtr adapter;
+				int adapterArraySize = 0;
+				if( colorSplineData )
+				{
+					knotOffset = getEndPointDuplication( colorSplineData->readable().basis ).first;
+					adapter = new Shader( "Utility/ColorArrayInputAdapter", "osl:shader" );
+
+					const Color3fVectorData *splineValuesData = findValues != newParameters.end() ? runTimeCast<const Color3fVectorData>( findValues->second.get() ) : nullptr;
+					if( !splineValuesData )
+					{
+						throw IECore::Exception( "Internal failure in convertToOSLConventions - expandSpline did not create values." );
+					}
+
+					const std::vector< Imath::Color3f > &splineValues = splineValuesData->readable();
+
+					adapterArraySize = splineValues.size();
+
+					if( adapterArraySize <= maxArrayInputAdapterSize )
+					{
+						for( unsigned int i = 0; i < splineValues.size(); i++ )
+						{
+							adapter->parameters()[ g_arrayInputNames[i] ] = new Color3fData( splineValues[i] );
+						}
+					}
+				}
+				else
+				{
+					knotOffset = getEndPointDuplication( floatSplineData->readable().basis ).first;
+					adapter = new Shader( "Utility/FloatArrayInputAdapter", "osl:shader" );
+
+					const FloatVectorData *splineValuesData = findValues != newParameters.end() ? runTimeCast<const FloatVectorData>( findValues->second.get() ) : nullptr;
+					if( !splineValuesData )
+					{
+						throw IECore::Exception( "Internal failure in convertToOSLConventions - expandSpline did not create values" );
+					}
+
+					const std::vector< float > &splineValues = splineValuesData->readable();
+
+					adapterArraySize = splineValues.size();
+
+					if( adapterArraySize <= maxArrayInputAdapterSize )
+					{
+						for( unsigned int i = 0; i < splineValues.size(); i++ )
+						{
+							adapter->parameters()[ g_arrayInputNames[i] ] = new FloatData( splineValues[i] );
+						}
+					}
+				}
+
+				if( adapterArraySize > maxArrayInputAdapterSize )
+				{
+					throw IECore::Exception(
+						"Cannot handle input to " + connection.destination.shader.string() + "." +
+						destName + ". Expanded spline has " + std::to_string( adapterArraySize) +
+						" control points, but max input adapter size is " + std::to_string( maxArrayInputAdapterSize )
+					);
+				}
+
+				// TODO - does this need to be tagged to be reversible?
+				// ANSWER - it's a special node type, which users shouldn't be creating, so John pointed out
+				// we can probably just remove them all without needing to tag
+				//adapter->blindData()->writable()[ componentConnectionAdapterLabel() ] = g_trueData;
```

Comment:
If this is the way we're going, then I think we need to change the Gaffer PR to make the adapter more hidden.